src/adr/0026-removing-configured-additional-mailboxListeners.md

```diff
@@ -0,0 +1,35 @@
+# 26. Removing a configured additional MailboxListener
+
+Date: 2020-04-03
+
+## Status
+
+Accepted (lazy consensus)
+
+## Context
+
+James enables a user to register additional mailbox listeners.
+
+The distributed James server is handling mailbox event processing (mailboxListener execution) using a RabbitMQ work-queue
+per listener.
+
+The distributed James server then declares a queue upon start for each one of these user registered listeners, that it
+binds to the main event exchange. 
+
+If the user unconfigures the listener, the queue and the binding are still present but not consumed. This result in 
+unbounded queue growth eventually causing RabbitMQ resource exhaustion and failure.
+
+## Decision
+
+We need, upon start, to sanitize bindings, and removes the ones corresponding to mailboxListeners that are not configured,
+using a name matching strategy.
+
+The queue should not be deleted to prevent message loss.
+
+Given a James topology with a non uniform configuration, the effective RabbitMQ routing will be the one of the latest 
+started James server.
+
+## Alternatives
+
+We could also considering adding a webadmin endpoint to sanitize eventBus bindings, allowing more predictability than the
```

Comment:
what about using EventSourcing to handle configuration changes regarding listeners?